knative-operator/hack/update-manifests.sh

```diff
@@ -41,48 +40,19 @@ function download_kafka {
   done
 }
 
-download_kafka eventing-kafka channel "$KNATIVE_EVENTING_KAFKA_VERSION" "${kafka_channel_files[@]}"
-
-# For 1.17 we still skip HPA
-git apply "$root/knative-operator/hack/001-eventing-kafka-remove_hpa.patch"
-
-# SRVKE-919: Change the minavailable pdb for kafka-webhook to 0
-git apply "$root/knative-operator/hack/007-eventing-kafka-patch-pdb.patch"
-
-# Kafka Broker content:
-# Control-Plane files:
-download_kafka eventing-kafka-broker controller "$KNATIVE_EVENTING_KAFKA_BROKER_VERSION" "${kafka_controller_files[@]}"
-
-#Data-Plane Files Broker:
-download_kafka eventing-kafka-broker broker "$KNATIVE_EVENTING_KAFKA_BROKER_VERSION" "${kafka_broker_files[@]}"
-
-#Data-Plane Files Sink:
-download_kafka eventing-kafka-broker sink "$KNATIVE_EVENTING_KAFKA_BROKER_VERSION" "${kafka_sink_files[@]}"
-
-#Data-Plane Files Source:
-download_kafka eventing-kafka-broker source "$KNATIVE_EVENTING_KAFKA_BROKER_VERSION" "${kafka_source_files[@]}"
-
-# That CM is already there, with Eventing
-git apply "$root/knative-operator/hack/001-broker-config-tracing.patch"
-
-# For now we remove the CRDs, since the "broker" does not yet do anything with them
-git apply "$root/knative-operator/hack/003-broker-remove-duplicated-crds.patch"
-
-# Remove the config for the new, unused channel
-git apply "$root/knative-operator/hack/004-remove_new_channel_cfg.patch"
-
-# Fix for SRVKE-1171
-git apply "$root/knative-operator/hack/011-eventing-kafkachannel-dead-letter-sink-uri.patch"
-git apply "$root/knative-operator/hack/012-eventing-kafkachannel-addressable-resolver-binding.patch"
-
-# SRVKE-1184 Migration to Broker components
-git apply "$root/knative-operator/hack/014-eventing-kafka-lease-name-remapping.patch"
-
-# SRVKE-1202 temporary fix for a missing ConfigMap knative-eventing/config-kafka-source-defaults
-git apply "$root/knative-operator/hack/017-eventing-kafka-configmap-knative-eventing-config-kafka-source-defaults.patch"
-
-# Tmp fix for channel post job
-git apply "$root/knative-operator/hack/015-eventing-kafkachannel-no-generated-name-for-post-install.patch"
-
-# SRVKE-1162: potential fix
-git apply "$root/knative-operator/hack/016-eventing-kafka-dispatcher-probes.patch"
+rm -rf "${component_dir}/controller"
+rm -rf "${component_dir}/broker"
+rm -rf "${component_dir}/channel"
+rm -rf "${component_dir}/source"
+rm -rf "${component_dir}/sink"
+
+download_kafka controller "${kafka_controller_files[@]}"
+download_kafka broker "${kafka_broker_files[@]}"
+download_kafka channel "${kafka_channel_files[@]}"
+download_kafka sink "${kafka_sink_files[@]}"
+download_kafka source "${kafka_source_files[@]}"
+
+# __Note__
+# artifacts are downloaded from midstream openshift/release/artifacts directory.
```

Comment:
was all of that really needed?

I'd hope that if we download from midstream: we do it consistent for all artifacts - and perhaps in a more dedicated PR 

Not against it .... but just a bit surprised

Reply:
I think it's needed now, going through all the patches and updating them for the new setup would have caused spending a lot of time on this, so now, it's much cleaner.

I get the concern but separate PRs in this case would have produced unneeded additional work and probably fewer cleanups.

Reply:
Ok, that;s a more than reasonable thing to do this - thanks for clarification